### PR TITLE
Add Apple Pay XML value objects for creating Auth request

### PR DESCRIPTION
**Pull request by timcwebster**
Create Apple Pay value objects used when generating the XML for an auth requestt.

### CODE REVIEW

File: features/integration/api/authorization.feature [hunk 1]
```diff
@@ -420,4 +420,42 @@ Feature: A payment authorization request is made against the Worldpay payment ga
             | shippingAddress     | Tim,Webster,7,Crisswell Close,Crownhill,,Milton Keynes,Bucks,GB,07426111111                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  |
         Then I should receive an authorised response
         And the response should be successful
-        And the response should contain "11" as card expiry month and "2020" as card expiry year
\ No newline at end of file
+        And the response should contain "11" as card expiry month and "2020" as card expiry year
+
+    Scenario: Generate Apple pay auth request xml
```
Comment (ubick) [inline features/integration/api/authorization.feature:425]: What's the value of checking XML conversion inside a Behat test compared to doing it via phpspec?

File: src/Inviqa/Worldpay/Api/Request/AuthorizeRequestFactory.php [hunk 2]
```diff
@@ -201,4 +235,60 @@ public function buildFromRequestParameters(array $parameters): PaymentService
...
         return $paymentService;
     }
+
+    /**
+     * @param array $parameters
+     *
+     * @return PaymentService
+     * @throws InvalidRequestParameterException
+     */
+    public function buildApplePayFromRequestParameters(array $parameters): PaymentService
```
Comment (ubick) [inline src/Inviqa/Worldpay/Api/Request/AuthorizeRequestFactory.php]: This class is now quite large.
Is it worth splitting it down into multiple factories? e.g. have a separate one just for the ApplePay parameters?

File: src/Inviqa/Worldpay/Api/Request/ApplePayPaymentService/MerchantCode.php [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+namespace Inviqa\Worldpay\Api\Request\ApplePayPaymentService;
+
+use Inviqa\Worldpay\Api\XmlAttributeDefaults;
+
+class MerchantCode extends XmlAttributeDefaults
```
Comment (ubick) [inline src/Inviqa/Worldpay/Api/Request/ApplePayPaymentService/MerchantCode.php]: A large number of value objects are the same across both standard Worldpay and ApplePay requests.
Is there any way of relying on the existing objects and only defining new ones for Apple, such as the Apple SSL ones?

**Review by ubick — CHANGES_REQUESTED** [review]
Great work.
Just added a few comments.

File: src/Inviqa/Worldpay/Api/Request/PaymentService/Submit/Authorisation/AuthorisationOrder.php [hunk 4]
```diff
@@ -28,7 +28,7 @@ public function __construct(
         OrderCode $orderCode,
         Description $description,
         Amount $amount,
-        PaymentDetails $paymentDetails,
+        PaymentDetailsApplePay $paymentDetails,
```
Comment (ubick) [inline src/Inviqa/Worldpay/Api/Request/PaymentService/Submit/Authorisation/AuthorisationOrder.php]: The standard authorization order shouldn't have any reference to Apple Pay

Comment (timcwebster) [reply src/Inviqa/Worldpay/Api/Request/PaymentService/Submit/Authorisation/AuthorisationOrder.php]: y find and replace fail

File: features/bootstrap/Services/OrderFactory.php [hunk 5]
```diff
@@ -19,7 +19,7 @@
 use Inviqa\Worldpay\Api\Request\PaymentService\Submit\Authorisation\Order\Param\Name;
 use Inviqa\Worldpay\Api\Request\PaymentService\Submit\Authorisation\Order\Param\Param;
 use Inviqa\Worldpay\Api\Request\PaymentService\Submit\Authorisation\Order\Param\Value as ParamValue;
-use Inviqa\Worldpay\Api\Request\PaymentService\Submit\Authorisation\Order\PaymentDetails;
+use Inviqa\Worldpay\Api\Request\PaymentService\Submit\Authorisation\Order\PaymentDetailsApplePay;
```
Comment (ubick) [inline features/bootstrap/Services/OrderFactory.php]: `PaymentDetailsApplePay` doesn't seem to be used in this factory?

File: src/Inviqa/Worldpay/Api/Request/AuthorizeRequestFactory.php [hunk 6]
```diff
@@ -2,6 +2,7 @@
 
 namespace Inviqa\Worldpay\Api\Request;
 
+use http\Header;
```
Comment (ubick) [inline src/Inviqa/Worldpay/Api/Request/AuthorizeRequestFactory.php]: Is this used?

File: src/Inviqa/Worldpay/Api/Request/AuthorizeRequestFactoryApplePay.php [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+        'ephemeralPublicKey' => "dummy public key",
+        'publicKeyHash' => "dummy hash",
+        'transactionId' => "123456",
+    ];
+
+    /**
+     * @param array $parameters
+     *
+     * @return PaymentService
+     * @throws \Inviqa\Worldpay\Api\Exception\InvalidRequestParameterException
```
Comment (ubick) [inline src/Inviqa/Worldpay/Api/Request/AuthorizeRequestFactoryApplePay.php]: namespace could be declared at the beginning of the file

**Review by ubick — APPROVED** [review]
LGTM :+1:

**Pull request merged**